File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]
        for i in range (2, numRows + 1):
            arr = [0] * i
            arr[0] = 1
            arr[i - 1] = 1
            for j in range(1, i - 1):
                arr[j] = triangle[i - 2][j - 1] +  triangle[i - 2][j]
            triangle.append(arr)
        return triangle

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        List<Integer> row = new ArrayList<>();
        
        row.add(1);
        triangle.add(row);
        int rowNum = 2;
        while (rowNum <= numRows) {
            row = new ArrayList<>();
            for (int i = 0; i < rowNum; i++) {
                if (i == 0 || i == rowNum - 1) {
                    row.add(1);
                } else {
                    row.add(triangle.get(rowNum - 2).get(i-1) + triangle.get(rowNum - 2).get(i));
                }
            }
            triangle.add(row);
            rowNum++;
        }
        
        return triangle;
    }
}